flink-formats/flink-avro/src/main/java/org/apache/flink/formats/avro/AvroRowDataSerializationSchema.java

```diff
@@ -64,61 +41,59 @@
 
 	private static final long serialVersionUID = 1L;
 
+	/** Nested schema to serialize the {@link GenericRecord} into bytes. **/
+	private final SerializationSchema<GenericRecord> nestedSchema;
+
 	/**
 	 * Logical type describing the input type.
 	 */
 	private final RowType rowType;
 
-	/**
-	 * Runtime instance that performs the actual work.
-	 */
-	private final SerializationRuntimeConverter runtimeConverter;
-
 	/**
 	 * Avro serialization schema.
 	 */
 	private transient Schema schema;
 
 	/**
-	 * Writer to serialize Avro record into a Avro bytes.
-	 */
-	private transient DatumWriter<IndexedRecord> datumWriter;
-
-	/**
-	 * Output stream to serialize records into byte array.
+	 * Runtime instance that performs the actual work.
 	 */
-	private transient ByteArrayOutputStream arrayOutputStream;
+	private final RowDataToAvroConverters.RowDataToAvroConverter runtimeConverter;
 
 	/**
-	 * Low-level class for serialization of Avro values.
+	 * Creates an Avro serialization schema with the given record row type.
 	 */
-	private transient Encoder encoder;
+	public AvroRowDataSerializationSchema(RowType rowType) {
```

Comment:
How about we remove this ctor? IMO the logic from this ctor should be only in the factory. I know that this class in theory is `PublicEvolving` but practically it is only usable from Table API through the factory. Therefore in my opinion it is safe to drop this ctor.

The same applies to `SerializationSchema`.

Reply:
I would prefer a constructor with default implementation.

Reply:
Why?

Reply:
Why not, the avro row data format is default to SE/DE avro without schema, if other formats what to customize something, use the correct constructor. Write the same code everywhere just makes no sense.